x-pack/plugins/integration_assistant/server/graphs/ecs/validate.ts

```diff
@@ -27,6 +27,8 @@ function extractKeys(data: AnyObject, prefix: string = ''): Set<string> {
       if ([...valueFieldKeys].every((k) => valueKeys.has(k))) {
         keys.add(fullKey);
       } else {
+        // Needs to be added, so that fields mapped directly under the data_stream key is detected
+        keys.add(fullKey);
```

Comment:
God spot, since we are filtering out from combinedSamples against ecsMapping, it does not matter if valueFieldsKeys exist or not, so removed the now redundant part, tested it against a few different formats.